src/modules/AieraChat/modals/AddSourceDialog/index.tsx

```diff
@@ -92,46 +93,56 @@ export function AddSourceDialog({
                                   ))
                                   .with({ status: 'success' }, ({ data }) =>
                                       data.events.length > 0 ? (
-                                          data.events.map(({ id, title }) => {
+                                          data.events.map(({ id, title, eventDate }) => {
                                               const sourceAdded = hasSource(
                                                   { targetId: id, targetType: 'event', title },
                                                   sources
                                               );
+                                              const toggleSource = sourceAdded
+                                                  ? () => onRemoveSource({ targetId: id, targetType: 'event', title })
+                                                  : () =>
+                                                        onAddSource({
+                                                            confirmed: true,
+                                                            targetId: id,
+                                                            targetType: 'event',
+                                                            title,
+                                                            date: eventDate,
```

Comment:
I'm surprised this works...don't you need to add `date` to the `Source` interface?